include/LightningJSON/LightningJSON.hpp

```diff
@@ -568,8 +589,23 @@ namespace LightningJSON
 			return JSONObject(JSONType::Empty);
 		}
 
-		static JSONObject FromString(string_view const& jsonStr)
+		static JSONObject FromString(char const* const jsonStr, size_t const length)
 		{
+			return FromString(std::string_view(jsonStr, length));
+		}
+
+		static JSONObject FromString(std::string const& jsonStr)
```

Comment:
This needs to be very clearly documented that the lifetime of the string passed in must exceed the lifetime of the JSONObject and must not be modified in any way during that time. It should be documented on the other two as well, but especially so on this one, since holding pointers to the innards of a std::string is very unexpected behavior.

Reply:
I think we just shouldn't have an API that takes std::string directly. If users have to call the function to convert it to a string_view, that by itself is a form of documentation in what we're doing with it.

Reply:
I'm ok with the char* overload since it lets people who are programming C-style use the class without having to create an intermediate class themselves. I just don't like std::string because std::string owns its own memory, and APIs that accept std::string are generally expected to make copies and take ownership of the memory themselves. Pointing at the context violates that expectation. Principle of least surprise.